Mathlib/Algebra/MonoidAlgebra/Defs.lean

```diff
@@ -574,6 +574,31 @@ theorem liftNC_smul [MulOneClass G] {R : Type*} [Semiring R] (f : k →+* R) (g
 
 end MiscTheorems
 
+section Linear
+
+variable {A : Type*} [Semiring k] [Semiring A] [Module k A]
+
+variable (k A) in
+/-- Interpret `single i` as a linear map (but with `MonoidAlgebra` in the type, rather than
+`Finsupp`). -/
+def lsingle (i : G) : A →ₗ[k] MonoidAlgebra A G :=
+  Finsupp.lsingle i
+
+lemma lsingle_def (i : G) :
+    lsingle k A i = Finsupp.lsingle i := rfl
+
```

Comment:
Just to re-iterate; this lemma is bad, as it unfolds across the API boundary. Given it seems you don't use it anyway, can you just delete it? (same for the `AddMonoidAlgebra` version).

Reply:
To be super clear, the new `lsingle` definition is a good idea 🙂.